src/level.c

```diff
@@ -189,8 +207,31 @@ void UpdateEnemy(Enemy *enemy, float delta, Obstacle obstacles[MAX_NUMBER_OF_OBS
             }
         }
     }
+    
+    // Lógica para determinar se o inimigo deve atacar
+    if (ShouldEnemyAttack()) {
+        enemy->attack.active = true;
+        enemy->attack.dimensions.x = enemy->dimensions.x - enemy->attack.dimensions.width;
+        enemy->attack.dimensions.y = enemy->dimensions.y;
+        enemy->attack.dimensions.width = 20; // Largura do ataque
+        enemy->attack.dimensions.height = enemy->dimensions.height;
+    }
+    
+     // Atualização do ataque do inimigo
+    if (enemy->attack.active) {
+        enemy->attack.dimensions.x -= positionDelta;
 
+        if (enemy->attack.dimensions.x + enemy->attack.dimensions.width < 0) {
+            enemy->attack.active = false;
+        }
+    }
+
+    // Renderização do inimigo e seu ataque
     DrawTexture(sprite, enemy->dimensions.x, enemy->dimensions.y, WHITE);
+
+    if (enemy->attack.active) {
+        DrawRectangleRec(enemy->attack.dimensions, RED);
+    }

```

Comment:
Inimigos não têm ataque.

```suggestion

    DrawTexture(sprite, enemy->dimensions.x, enemy->dimensions.y, WHITE);
```